runtime/helpers/waitHelpers.js

```diff
@@ -1,67 +1,279 @@
+	await element.waitForEnabled(timeout, true, msg, mediumInterval);
+};
+
+async function waitUntilElementContainsText(selectorOrElement, expectedText, timeout = elementContainsTextTimeout) {
+	const element = await waitUntilElementIsPresent(selectorOrElement);
+	let textTimeoutMsg =
+		'Text: "' + expectedText + '" is not visible for element: "' + element.selector + '" within time: ' + timeout;
+	let actualText = "";
+	try {
+		await driver.waitUntil(
+			async () => {
+				actualText = await element.getText();
+				return actualText.includes(expectedText);
+			},
+			timeout,
+			textTimeoutMsg,
+			mediumInterval
+		);
+	} catch (error) {
+		const msg = error.message + "\n" + '" , Actual text: "' + actualText + '"';
+		throw msg;
+	}
+	return element;
+};
+
+async function waitUntilUrlContains(expectedUrlText, timeout = urlContainsTimeout) {
+	let textTimeoutMsg = 'Url does not contains text : "' + expectedUrlText + '" within time: ' + timeout;
+	let actualUrl = "";
+	try {
+		await driver.waitUntil(
+			async () => {
+				actualUrl = await driver.getUrl();
+				const urlContainsText = actualUrl.includes(expectedUrlText);
+				return urlContainsText;
+			},
+			timeout,
+			textTimeoutMsg,
+			mediumInterval
+		);
+	} catch (error) {
+		const msg = error.message + "\n" + '" , Actual URL: "' + actualUrl + "'";
+		throw msg;
+	}
+};
+
+async function waitUntilUrlNotContains(notExpectedUrlText, timeout = urlContainsTimeout) {
+	let textTimeoutMsg = 'URL should not contains text : "' + notExpectedUrlText + '" within time: ' + timeout;
+	let actualUrl = "";
+	try {
+		await driver.waitUntil(
+			async () => {
+				actualUrl = await driver.getUrl();
+				const urlContainsText = actualUrl.includes(notExpectedUrlText);
+				return !urlContainsText;
+			},
+			timeout,
+			textTimeoutMsg,
+			mediumInterval
+		);
+	} catch (error) {
+		const msg = error.message + "\n" + '"Actual URL: "' + actualUrl + "'";
+		throw msg;
+	}
+};
+
+async function waitUntilPageLoads(timeout = pageLoadingTimeout) {
+	const timeoutMsg = "Page is not loaded";
+	await waitUntilScriptResultIsTrue(() => document.readyState.includes("complete"), timeoutMsg, timeout);
+};
+
+async function waitUntilScriptResultIsTrue(script, timeoutMsg,  timeout = pageLoadingTimeout) {
+	await driver.waitUntil(
+		async () => {
+			const result = await driver.execute(script);
+			return result;
+		},
+		timeout,
+		timeoutMsg + "\n" + "Script: " + script + " result: " + result,
+		shortInterval
+	);
+};
+
+async function waitAndClick(selectorOrElement) {
+	let  element = await waitUntilElementIsPresent(selectorOrElement);
+	await waitUntilElementIsClickable(element);
+	await element.click();
+	//Pause is temporary: should be removed in SC-6661
+	await driver.pause(1000);
+	await waitUntilPageLoads();
+	
+};
+
+async function waitAndSetValue(selectorOrElement, expectedValue, timeout = MEDIUM_WAIT_MILLIS) {
```

Comment:
I only have 2 general comments,
1. Do we need to keep these timeouts in the parameter list in functions as it looks a bit bad and gets unreadable and long, we can put them inside a function
2. It's nice to have the refreshPage function, but we can implement it later

Reply:
1. As we have talked, this is the desired effect. timeout = value -> this is default value.
Thanks for this we can use this method in to ways:
waitAndSetValue(selectorOrElement, expectedValue)  -> default  timeout 
waitAndSetValue(selectorOrElement, expectedValue, 120000)
2. Only waitHelpers were the goal of my task. This could be implemented later. When needed.